js/csv.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * Converts a JSON object to a valid CSV string file.
+ *
+ * @param {Object} object The object to convert.
+ * @returns {String} The converted string.
+ */
+export const objectToCsv = object => {
+    let csv = "";
+    if (Object.keys(object) === 0) return csv;
+
+    const fields = Object.keys(object[0]);
+    csv += fields.toString() + "\n";
+
+    for (let i = 0; i < Object.keys(object).length; i++) {
+        let entry = "";
+        for (let j = 0; j < fields.length; j++) {
+            // values containing the ',' character need to be parsed,
+            // otherwise values will not align with fields
+            const value = object[i][fields[j]];
+            if (String(value).includes(",")) {
+                entry += '"' + String(value).replace(/["']/g, "") + '",';
+            } else {
+                // only append truish values
+                entry += value ? value + "," : ",";
+            }

```

Comment:
Can't you use the `_toString` at the bottom of the file?

Reply:
Haven't tried this, as I wasn't aware of these functions. I'll test them out, and if they present the same result, I'll change it to use the existing function.

Reply:
Upon further testing, the `_toString` doesn't do quite what was necessary, and in the case of this conversion it results in an incorrect parsing. In this case, this if doesn't just convert to string, it also ensures that even null values appear, so as to not format the object incorrectly.